$OpenBSD: patch-CMakeLists_txt,v 1.2 2018/01/09 23:03:00 bcallah Exp $

Don't do this git stuff.
Don't do the cpplint stuff. Elimintates python as a dependency.

Index: CMakeLists.txt
--- CMakeLists.txt.orig
+++ CMakeLists.txt
@@ -7,34 +7,10 @@ set(PROJECT_VERSION "0.2.0")
 
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
 
-find_package(Git)
-if(GIT_FOUND)
-  execute_process(COMMAND ${GIT_EXECUTABLE} describe --always --tags --dirty
-                  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-                  OUTPUT_VARIABLE FREESERF_VERSION
-                  OUTPUT_STRIP_TRAILING_WHITESPACE)
-  string(FIND ${FREESERF_VERSION} "." VERSION_FULL)
-  if(VERSION_FULL STREQUAL "-1")
-    string(REGEX REPLACE "^([0-9]+)\\..*" "\\1" PROJECT_VERSION_MAJOR "${PROJECT_VERSION}")
-    string(REGEX REPLACE "^[0-9]+\\.([0-9]+).*" "\\1" PROJECT_VERSION_MINOR "${PROJECT_VERSION}")
-    execute_process(COMMAND ${GIT_EXECUTABLE} rev-parse --short=7 HEAD
-                    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-                    OUTPUT_VARIABLE PROJECT_VERSION_PATCH
-                    OUTPUT_STRIP_TRAILING_WHITESPACE)
-    set(FREESERF_VERSION "${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}")
-    set(PROJECT_VERSION "${FREESERF_VERSION}")
-  else()
-    string(REGEX REPLACE "^v([0-9]+)\\..*" "\\1" PROJECT_VERSION_MAJOR "${FREESERF_VERSION}")
-    string(REGEX REPLACE "^v[0-9]+\\.([0-9]+).*" "\\1" PROJECT_VERSION_MINOR "${FREESERF_VERSION}")
-    string(REGEX REPLACE "^v[0-9]+\\.[0-9]+.*-g([0-9a-f]+).*" "\\1" PROJECT_VERSION_PATCH "${FREESERF_VERSION}")
-    set(PROJECT_VERSION "${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}")
-  endif()
-else()
   set(FREESERF_VERSION "${PROJECT_VERSION}")
   string(REGEX REPLACE "^([0-9]+)\\..*" "\\1" PROJECT_VERSION_MAJOR "${PROJECT_VERSION}")
   string(REGEX REPLACE "^[0-9]+\\.([0-9]+).*" "\\1" PROJECT_VERSION_MINOR "${PROJECT_VERSION}")
   string(REGEX REPLACE "^[0-9]+\\.[0-9]+\\.([0-9]+).*" "\\1" PROJECT_VERSION_PATCH "${PROJECT_VERSION}")
-endif()
 
 message(STATUS "FREESERF_VERSION = ${FREESERF_VERSION}")
 message(STATUS "PROJECT_VERSION = ${PROJECT_VERSION}")
@@ -55,9 +31,6 @@ configure_file(${CMAKE_CURRENT_SOURCE_DIR}/src/version
                @ONLY)
 
 add_definitions(-DPACKAGE_BUGREPORT="https://github.com/freeserf/freeserf/issues")
-
-include(CppLint)
-enable_check_style()
 
 add_subdirectory(src)
 
